calico_kubernetes/calico_kubernetes_cni.py

```diff
@@ -103,30 +131,29 @@ def delete(args):
                       container_id=container_id)
 
     # Delete profile if only member
-    if datastore_client.profile_exists(net_name) and \
-       len(datastore_client.get_profile_members(net_name)) < 1:
+    if _datastore_client.profile_exists(profile_name) and \
+       len(_datastore_client.get_profile_members(profile_name)) < 1:
         try:
-            _log.info("Profile %s has no members, removing from datastore" % net_name)
-            datastore_client.remove_profile(net_name)
+            _log.info("Profile %s has no members, removing from datastore" % profile_name)
+            _datastore_client.remove_profile(profile_name)
         except:
             _log.error("Cannot remove profile %s: Profile cannot be found." % container_id)
             sys.exit(1)
 
 
-def _create_calico_endpoint(container_id, interface, subnet):
+def _create_calico_endpoint(container_id, interface):
```

Comment:
Is this wrapper necessary? Seems better to just move the endpoint check into `_container_add` and removing this function.